src/storage/storage_util.cpp

```diff
@@ -0,0 +1,100 @@
+#include <unordered_map>
+#include "storage/storage_util.h"
+#include "storage/tuple_access_strategy.h"
+#include "storage/delta_record.h"
+
+namespace terrier::storage {
+void StorageUtil::WriteBytes(uint8_t attr_size, uint64_t val, byte *pos) {
+  switch (attr_size) {
+    case sizeof(uint8_t):*reinterpret_cast<uint8_t *>(pos) = static_cast<uint8_t>(val);
+      break;
+    case sizeof(uint16_t):*reinterpret_cast<uint16_t *>(pos) = static_cast<uint16_t>(val);
+      break;
+    case sizeof(uint32_t):*reinterpret_cast<uint32_t *>(pos) = static_cast<uint32_t>(val);
+      break;
+    case sizeof(uint64_t):*reinterpret_cast<uint64_t *>(pos) = static_cast<uint64_t>(val);
+      break;
+    default:
+      // Invalid attr size
+      throw std::runtime_error("Invalid byte write value");
+  }
+}
+
+uint64_t StorageUtil::ReadBytes(uint8_t attr_size, const byte *pos) {
+  switch (attr_size) {
+    case sizeof(uint8_t):return *reinterpret_cast<const uint8_t *>(pos);
+    case sizeof(uint16_t):return *reinterpret_cast<const uint16_t *>(pos);
+    case sizeof(uint32_t):return *reinterpret_cast<const uint32_t *>(pos);
+    case sizeof(uint64_t):return *reinterpret_cast<const uint64_t *>(pos);
+    default:
+      // Invalid attr size
+      throw std::runtime_error("Invalid byte write value");
+  }
+}
+
+void StorageUtil::CopyWithNullCheck(const byte *from, ProjectedRow *to, uint8_t size, uint16_t col_id) {
+  if (from == nullptr)
+    to->SetNull(col_id);
+  else
+    WriteBytes(size, ReadBytes(size, from), to->AccessForceNotNull(col_id));
+}
+
+void StorageUtil::CopyWithNullCheck(const byte *from,
+                                    const TupleAccessStrategy &accessor,
+                                    TupleSlot to,
+                                    uint16_t col_id) {
+  if (from == nullptr) {
+    accessor.SetNull(to, col_id);
+  } else {
+    uint8_t size = accessor.GetBlockLayout().attr_sizes_[col_id];
+    WriteBytes(size, ReadBytes(size, from), accessor.AccessForceNotNull(to, col_id));
+  }
+}
+
+void StorageUtil::CopyAttrIntoProjection(const TupleAccessStrategy &accessor,
```

Comment:
`void StorageUtil::CopyAttrIntoProjection(const TupleAccessStrategy &accessor,
                                          const TupleSlot from,
                                          ProjectedRow *const to,
                                          const uint16_t projection_list_offset)`